bench/benchmarks/fps.js

```diff
@@ -0,0 +1,100 @@
+var measureFramerate = require('../lib/measure_framerate');
+
+var smallFeatureCollection = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "properties": {},
+        "geometry": {
+            "type": "Point",
+            "coordinates": [ -77.032194, 38.912753 ]
+        }
+    }]
+};
+
+var largeFeatureCollection = require('../data/naturalearth-land.json');
+
+var DURATION_MILLISECONDS = 5 * 1000;
+
+module.exports = function(options) {
+    var evented = util.extend({}, Evented);
+
+    var map = options.createMap({
+        width: 1024,
+        height: 768,
+        zoom: 5,
+        center: [-77.032194, 38.912753],
+        style: 'mapbox://styles/mapbox/bright-v8'
+    });
+
+    map.on('load', function() {
+        map.addSource('geojson', {
+            'type': 'geojson',
+            'data': {
+                'type': 'FeatureCollection',
+                'features': []
+            }
+        });
+
+        map.addLayer({
+            'id': 'geojson-polygon',
+            'source': 'geojson',
+            'type': 'fill',
+            'filter': ['all', ['==', '$type', 'Polygon']],
+            'paint': {
+                'fill-color': '#000000',
+                'fill-opacity': 0.25
+            }
+        });
+
+        map.addLayer({
+            'id': 'geojson-point',
+            'source': 'geojson',
+            'type': 'circle',
+            'filter': ['all', ['==', '$type', 'Point']],
+            'paint': {
+                'circle-radius': 5,
+                'circle-color': '#000000'
+            }
+        });
+
+        var source = map.getSource('geojson');
+
+        var testSetData = function(geojson, cb) {
+            var running = true;
+            requestAnimationFrame(function runner() {
+                if (running) {
+                    source.setData(geojson);
+                    requestAnimationFrame(runner);
+                }
+            });
+
+            measureFramerate(DURATION_MILLISECONDS, function(err, fps) {
+                running = false;
+                cb(err, fps);
+            });
+        };
+
+        map.repaint = true;
+        measureFramerate(DURATION_MILLISECONDS, function(err, fps) {
+            if (err) return evented.fire('error', {error: err});
+            evented.fire('log', {message: 'repaint - ' + formatNumber(fps)});
+            map.repaint = false;
+            testSetData(smallFeatureCollection, function(err, fps) {
+                if (err) return evented.fire('error', {error: err});
+                evented.fire('log', {message: 'setData (small) - ' + formatNumber(fps)});
+                testSetData(largeFeatureCollection, function(err, fps) {
+                    if (err) return evented.fire('error', {error: err});
+                    evented.fire('log', {message: 'setData (large) - ' + formatNumber(fps)});
+                });
+            });
+        });
+    });
+
+    return evented;
+};
```

Comment:
Make sure you're adhering to the benchmark spec, defined in [bench/README.md](https://github.com/mapbox/mapbox-gl-js/blob/master/bench/README.md#end). This was spec designed to allow us to automate benchmark runs in the near future.

 If the benchmark doesn't have a "single" end score because you're measuring several different things, you can break the benchmark up into multiple benchmarks.